otx/algorithms/detection/adapters/mmdet/evaluation/evaluator.py

```diff
@@ -0,0 +1,393 @@
+    ious_argmax = ious.argmax(axis=1)
+    # sort all dets in descending order by scores
+    sort_inds = np.argsort(-cls_scores)
+
+    gt_covered = np.zeros(num_gts, dtype=bool)
+    # if no area range is specified, gt_area_ignore is all False
+    for i in sort_inds:
+        if ious_max[i] >= iou_thr:
+            matched_gt = ious_argmax[i]
+            if not gt_covered[matched_gt]:
+                gt_covered[matched_gt] = True
+                gt_covered_iou[matched_gt] = ious_max[i]
+                tp[i] = 1
+            else:
+                fp[i] = 1
+            # otherwise ignore this detected bbox, tp = 0, fp = 0
+        else:
+            fp[i] = 1
+    return tp, fp, np.mean(gt_covered_iou)
+
+
+class Evaluator:
+    """OTX Evaluator for mAP and mIoU.
+
+    Args:
+            annotation (list(dict)): ground truth annotation
+            domain (Domain): OTX algorithm domain
+            classes (list): list of classes
+            nproc (int, optional): number of processes. Defaults to 4.
+    """
+
+    def __init__(self, annotation: List[Dict], domain: Domain, classes: List[str], nproc=4):
+        self.domain = domain
+        self.classes = classes
+        self.num_classes = len(classes)
+        if domain != Domain.DETECTION:
+            self.annotation = self.get_gt_instance_masks(annotation)
+        else:
+            self.annotation = annotation
+        self.nproc = nproc
+
+    def get_gt_instance_masks(self, annotation: List[Dict]):
+        """Format ground truth instance mask annotation.
+
+        Args:
+            annotation (List[Dict]): per-image ground truth annotation
+
+        Returns:
+            cls_anno_list: per-class ground truth instance mask list
+        """
+        num_images = len(annotation)
+        cls_anno_list: List[List] = [[] for _ in range(self.num_classes)]
+        for idx in range(num_images):
+            masks = annotation[idx]["masks"]
+            for class_id in range(self.num_classes):
+                gt_inds = annotation[idx]["labels"] == class_id
+                polygon_masks = []
+                if gt_inds.any():
+                    gt_inds = np.where(gt_inds == 1)[0]
+                    polygon_masks = masks[gt_inds]
+                cls_anno_list[class_id].append(polygon_masks)
+        return cls_anno_list
+
+    def get_mask_det_results(self, det_results: List[Tuple], class_id: int) -> Tuple[List, List]:
+        """Get mask detection results for a specific class.
+
+        Args:
+            det_results (list(tuple)): detection results including bboxes and masks
+            class_id (int): class index
+
+        Returns:
+            cls_dets: per-class detection results including bboxes and decoded masks
+            cls_scores: class scores
+        """
+        cls_scores = [img_res[0][class_id][..., -1] for img_res in det_results]
+        cls_dets: List[Tuple] = []
+        for i, det in enumerate(det_results):
```

Comment:
```suggestion
        for det in det_results:
```
I think i isn't used here